### PR TITLE
[MINOR] improve(script): Support collect the output stream into out file

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes were proposed in this pull request?

Collect the standard out and error stream content into `xxx.out` file.

### Why are the changes needed?

It can really help us for troubleshooting.  There are some exception throws but not caught, the call stack should be print in the `out` file. 

Reference what hadoop, alluxio did.

### Does this PR introduce _any_ user-facing change?

Supply a out file.

### How was this patch tested?

```console
$ bin/start-dashboard.sh  
$ cat logs/dashboard.out                   
-XX:ConcGCThreads=5 -XX:InitialHeapSize=8589934592 -XX:InitiatingHeapOccupancyPercent=45 -XX:MaxGCPauseMillis=200 -XX:MaxHeapSize=8589934592 -XX:ParallelGCThreads=20 -XX:+PrintAdaptiveSizePolicy -XX:+PrintCommandLineFlags -XX:+PrintGC -XX:+PrintGCApplicationStoppedTime -XX:+PrintGCCause -XX:+PrintGCDateStamps -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintPromotionFailure -XX:+PrintTenuringDistribution -XX:+UseCompressedClassPointers -XX:+UseCompressedOops -XX:+UseG1GC 
```

### CODE REVIEW

File: bin/start-shuffle-server.sh [hunk 1]
```diff
@@ -149,7 +150,7 @@ else
 fi
 
 SHUFFLE_SERVER_JAVA_OPTS=${SHUFFLE_SERVER_JAVA_OPTS:-""}
-$RUNNER ${SHUFFLE_SERVER_BASE_JVM_ARGS} ${SHUFFLE_SERVER_JVM_GC_ARGS} ${JVM_LOG_ARGS} ${JAVA_LIB_PATH} ${SHUFFLE_SERVER_JAVA_OPTS} -cp ${CLASSPATH} ${MAIN_CLASS} --conf "${SHUFFLE_SERVER_CONF_FILE}" $@ &
+(nohup $RUNNER ${SHUFFLE_SERVER_BASE_JVM_ARGS} ${SHUFFLE_SERVER_JVM_GC_ARGS} ${JVM_LOG_ARGS} ${JAVA_LIB_PATH} ${SHUFFLE_SERVER_JAVA_OPTS} -cp ${CLASSPATH} ${MAIN_CLASS} --conf "${SHUFFLE_SERVER_CONF_FILE}" $@ > ${LOG_OUT_PATH} 2>&1) &
```
Comment (zuston) [inline bin/start-shuffle-server.sh:153]: I'm OK for this improvement. But I remember that this had been supported in the previous PRs. Maybe someone deleted this. Could you help check this? @maobaolong

Comment (maobaolong) [reply bin/start-shuffle-server.sh:153]: @zuston I found it!
- Introduced by your PR #226 
- Removed by @rickyma ‘s  PR #1553

Comment (zuston) [reply bin/start-shuffle-server.sh:153]: Got it. LGTM now.

**Review by zuston — APPROVED** [review]
lgtm

**Pull request merged**